prog/dftb+/lib_dftbplus/mainio.F90

```diff
@@ -3743,8 +3744,7 @@ end subroutine printSccInfo
 
 
   !> Prints info about scc convergence.
-  subroutine printReksSccInfo(iSccIter, Etotal, diffTotal, &
-      & sccErrorQ, t1, t2, FONs, tSSR22, tSSR44)
+  subroutine printReksSccInfo(iSccIter, Etotal, diffTotal, sccErrorQ, deltaT, reks)
```

Comment:
Originally my purpose for this file and whole codes is that the the `reks` type variables is not used except interface routines. Thus, this variable is not added to lib_reks/*.F90 files except reksinterface.F90. Likewise I did not add the `reks` variable in this file (mainio.F90), but you did. Is this more nice/well-organized coding for DFTB+?

Reply:
We are in the process of changing to simplify the main code by passing derived types into subroutines, instead of components of a type. There are two reasons for this:
1) The higher level code becomes simpler as details of what the data type contains is only needed to be known inside the routines directly processing that type, not the higher level code.
2) It makes the calling arguments to routines shorter in the higher level code, but more importantly it is less likely to be broken by other developers (who are less familiar with the routines) if they change the code. This is since the compile time checking can just examine the derived type position in the argument list, which would be different from other variables the calling list. If the individual components are passed, there is a higher chance of requiring positional orderings of variables with the same signature to be kept correct.

The big disadvantage, as I'm sure you know, is that if parts of the derived type should be intent in and others intent out,  then the whole type needs to be intent (inout). We are in the process of working out a solution for part of this problem with use of protected and another mechanism (at least for large arrays) to allow read only contents from pointers (the current Fortran standard does not offer a mechanism to do this). But this is unlikely to be decided on before the next code release.

Reply:
Thanks for your answers. Then, should I change the variables used in the routines of lib_reks/*.F90 files for simplification?

Reply:
Yes, that would be very helpful, thank you.

Reply:
Then, I will modify the parts related to the type `reks` variables in the whole codes. Please check the commits in the later.